R/remote.R

```diff
@@ -0,0 +1,58 @@
+
+isChildProcess <- function() {
+  !is.na(Sys.getenv("RSTUDIOAPI_IPC_REQUESTS_FILE", unset = NA))
+}
+
+callRemote <- function(call, frame) {
+
+  # check for active request / response
+  request  <- Sys.getenv("RSTUDIOAPI_IPC_REQUESTS_FILE", unset = NA)
+  response <- Sys.getenv("RSTUDIOAPI_IPC_RESPONSE_FILE", unset = NA)
+  secret   <- Sys.getenv("RSTUDIOAPI_IPC_SHARED_SECRET", unset = NA)
+  if (is.na(request) || is.na(response) || is.na(secret))
+    stop("internal error: callFunRemote() called without remote connection")
+
+  attr(call, "srcref") <- NULL
+
+  # ensure rstudioapi functions get appropriate prefix
+  if (is.name(call[[1L]]))
+    call[[1L]] <- call("::", as.name("rstudioapi"), call[[1L]])
+
+  # ensure arguments are evaluated before sending request
+  for (i in seq_along(call)[-1L])
+    call[[i]] <- eval(call[[i]], envir = frame)
+
+  # write to tempfile and rename, to ensure atomicity
+  data <- list(secret = secret, call = call)
+  tmp <- tempfile(tmpdir = dirname(request))
+  saveRDS(data, file = tmp)
+  file.rename(tmp, request)
+
+  # loop until response is ready (poll)
```

Comment:
> It seems unlikely that users would want to make a large number of rstudioapi calls in succession from one script

I do think this is going to be common. One use of this feature will be to have child R sessions reporting progress to the parent session; for example, knitr could detect when it's being run as a child RStudio process and use rstudioapi to draw progress bars in RStudio as the knit proceeds, instead of its current ASCII bars. knitr can emit dozens of updates in the space of a second; waiting for all of the IPC files to sync and clear will cause the progress bar update time to exceed the knit time! 

That said, we could take this as-is now and implement async later if we need it.